docs/updater.md

```diff
@@ -0,0 +1,102 @@
+
+```
+mUpdater->on(
+      gvk::swapchain_format_changed_event(gvk::context().main_window()),
+      gvk::swapchain_additional_attachments_changed_event(gvk::context().main_window())
+   ).invoke([this]() {
+      // update render pass here
+   }).then_on(
+      gvk::swapchain_changed_event(gvk::context().main_window()),
+      gvk::shader_files_changed_event(mPipeline)
+   ).update(mPipeline);
+```
+
+In the above example, the renderpass belonging to `mPipeline` requires recreation only if the events `swapchain_format_changed_event` or `swapchain_format_changed_event` arise. Then it is evaluated whether _any type_ of swapchain recreation has occured and if so, subsequently the pipeline is updated. It is important to note here: The evaluation of events given to `then_on()` is only temporaly restricted by the previous list: this evaluation will occur in any case.
+
+## Common Use-Cases
+* Swapchain recreation can lead to changes to image properties. This may cause any pipeline, renderpass, framebuffer or image view that directly works with swapchain to end in an invalid state. The updater mechanism is meant to streamline the recreation process of those objects, as well as to allow any user-defined behaviour to execute in cases where such events occur.
+* Shader hot reloading: Update graphics/compute/ray-tracing pipelines after (SPIR-V) shader files have been updated on the file system.
+* Adapt to number of concurrent frame changes: setup any necessary sync mechanism.
+
+## Notes on Swapchain recreation
+
+* Modifying the following attributes may trigger a swapchain recreation:
+  * window size,
+  * presentation mode,
+  * number of presentation images,
+  * number of concurrent frames,
+  * backbuffer image format,
+  * out of data, or suboptimal swapchain detection by vulkan.
+* The swapchain can only resize if the `window` has been configured to allow being resized (via `window::enable_resizing`).   
+
+## General Notes
+* The updater mechanism is invoked just before the `render()` call of the associated `gvk::invokee`.
+* When you hand over resources that shall be automatically updated to the `gvk::updater`, the `gvk::updater` needs to take ownership of them. Therefore, you'll have to pay the price of enabling shared resource ownership (by the means of `avk::owning_resource::enable_shared_ownership`) for all of these resources.
+* _Note about shader files being changed on the file system:_ The loaded shader files are watches for changes, i.e. the SPIR-V versions of shader files in the target directory. The most convenient way to get them updated is to leave the _Post Build Helper_ running in the background. Ensure that its setting "Do not monitor files during app execution" is _not_ enabled, and the _Post Build Helper_ will automatically compile shader files to SPIR-V at runtime if it detects changes to the original shader source files.
+* To use an updater independently, a new subtype of `gvk::invokee` can be created.
```

Comment:
What do you mean by that?

Reply:
BTW: This comment refers to the very last line (L#93) only. From the way this is displayed here on GitHub, this might not be clear since 3 more lines are shown in the view that is presented to me.

Reply:
I just wanted to point out that if for whatever reason someone just wants to listen to events, they can inherent from invokee and create some sort of event listening. It was not so clear, and probably not really used anywys, so I removed it.